src/NovelAudioService.cpp

```diff
@@ -0,0 +1,162 @@
+//
+// Created by krjoh on 29/08/2019.
+//
+
+#include "NovelAudioService.h"
+#include <iostream>
+#include <functional>
+
+namespace NovelRT {
+
+NovelAudioService::NovelAudioService(){
+  initializeAudio();
+  _nextChannel = 0;
+}
+
+bool NovelAudioService::initializeAudio(){
+  if (SDL_Init(SDL_INIT_AUDIO) < 0 )
```

Comment:
It's been a while since I've looked at SDL, but don't you need to call `SDL_Init(0)` somewhere and then use `SDL_InitSubSystem(flags)` if you want to initialize things (like audio vs graphics) separately?

Reply:
After review, it looks like `SDL_Init()` just makes the subsystem call after setting up crash handling, etc. Probably minimal but additional overhead thats unnecessary, so I'll revise this.

Reply:
Fixed, ty